pkg/gridtypes/zos/gw_fqdn.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 var (
-	gwNameRegex = regexp.MustCompile(`^[a-zA-Z0-9-.]+$`)
+	gwNameRegex = regexp.MustCompile(`^[a-zA-Z0-9-_.]+$`)
```

Comment:
There is a big issue here that is not related to your fix. It's nice that you add the `_` that was indeed a bug. but the bigger problem here is that the same regex is used for both NAME and FQDN validation which is not right! 

A name must not have a `.` (dot) because a name is going to be suffixed by the node domain. So a name can be `mario` or `mario-bassem` but NOT `mario.bassem`.

On the other hand a FQDN must be a valid domain name hence again this regex is invalid because this allow the dot to appear ANY where in the domain name like `.mario.bassem` or `mario..bassem` or `marrio.bassem..` which are all wrong.

IMHO nameRegex should be `^[a-zA-Z0-9-_]+$` same as above but without the dot
the fqdnRegex then should be  `^([a-zA-Z0-9-_]+\.)+[a-zA-Z0-9-_]{2,}$`
the fqdnRegex means the pattern for the name can be repeated multiple times separated by `.` but must end up with a min of 2 characters. so `mario.bassem.com` works but `mario.bassem.` or `mario.basse.m` won't work